SpatialGDK/Build/Scripts/BuildWorker.bat

```diff
@@ -1,22 +1,73 @@
 @echo off
 
-rem Generate worker configs
-pushd "%~dp0..\..\..\..\..\..\spatial"
-spatial worker build build-config
-popd
+if [%4] == [] goto :MissingParams
 
-rem Build Unreal project
-pushd "%~dp0..\..\..\..\..\"
+rem Try and build as a project plugin first, check for a project plugin structure.
+set UnrealProjectDir="%~dp0..\..\..\..\..\"
+set FoundUProject=""
 
+for /f "delims=" %%A in (' powershell -Command "(Get-ChildItem -Path "%UnrealProjectDir%" *.uproject).FullName" ') do set FoundUProject="%%A"
+
+if %FoundUProject%=="" (
+	goto :BuildAsEnginePlugin
+)
+
+rem If we are running as a project plugin then this will be the path to the spatial directory.
+echo Building as project plugin
+set SpatialDir="%~dp0..\..\..\..\..\..\spatial"
 set BUILD_EXE_PATH="Plugins\UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\Build.exe"
+goto :Build
+
+
+:BuildAsEnginePlugin
+rem If we are running as an Engine plugin then we need a full path to the .uproject file!
+set UPROJECT=%4
+
+if not exist %UPROJECT% (

```

Comment:
This won't get his anymore right?

Reply:
It would if you specify a non-existent uproject path.

Reply:
Actually this is a necessary check, essentially if you are using an Engine Plugin you must be using a full path to the `.uproject`. If the `.uproject` doesn't exist (meaning a relative path has been passed). Then we print the error below :D